# Import Package

In [206]:
import pandas as pd
from sklearn.model_selection import train_test_split

# Import Data

In [154]:
df = pd.read_csv("data_machine.csv", index_col='Unnamed: 0')
df.head()

vendor  myct  mmin   mmax  cach  chmin  chmax  prp
0  adviser   125   256   6000   256     16    128  198
1   amdahl    29  8000  32000    32      8     32  269
2   amdahl    29  8000  32000    32      8     32  220
3   amdahl    29  8000  32000    32      8     32  172
4   amdahl    29  8000  16000    32      8     16  132

# Melihat Jumlah Baris dan Kolom

In [155]:
df.shape

(209, 8)

terdiri dari 209 baris dan 8 kolom<br>
terdiri dari 7 fitur dan 1 target<br>

# Melihat Data yang Kosong

In [156]:
df.isnull().sum()

vendor    0
myct      0
mmin      0
mmax      0
cach      0
chmin     0
chmax     0
prp       0
dtype: int64

# Mengisi kolom yang kosong dengan nilai rata-rata dari kolom yang kosong jika Numeric

In [157]:
df.fillna(df['horsepower'].mean(), inplace=True)

C:\Users\RAIHAN~1\AppData\Local\Temp/ipykernel_11564/820435583.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  df.fillna(df.mean(), inplace=True)


In [158]:
df.isnull().sum()

vendor    0
myct      0
mmin      0
mmax      0
cach      0
chmin     0
chmax     0
prp       0
dtype: int64

In [159]:
df.vendor.value_counts()

ibm             32
nas             19
ncr             13
sperry          13
honeywell       13
siemens         12
cdc              9
amdahl           9
burroughs        8
hp               7
harris           7
dg               7
dec              6
c.r.d            6
ipl              6
magnuson         6
cambex           5
prime            5
formation        5
perkin-elmer     3
gould            3
nixdorf          3
wang             2
bti              2
basf             2
apollo           2
microdata        1
four-phase       1
sratus           1
adviser          1
Name: vendor, dtype: int64

In [170]:
df['vendor'].fillna('ibm', inplace=True)

tidak terdapat missing value atau data yang kosong

In [173]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 209 entries, 0 to 208
Data columns (total 8 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   vendor  209 non-null    int32
 1   myct    209 non-null    int64
 2   mmin    209 non-null    int64
 3   mmax    209 non-null    int64
 4   cach    209 non-null    int64
 5   chmin   209 non-null    int64
 6   chmax   209 non-null    int64
 7   prp     209 non-null    int64
dtypes: int32(1), int64(7)
memory usage: 13.9 KB


In [172]:
df.dtypes

vendor    int32
myct      int64
mmin      int64
mmax      int64
cach      int64
chmin     int64
chmax     int64
prp       int64
dtype: object

jumlah kolom dengan int64 sebanyak 7 kolom<br>
jumlah kolom float64 sebanyak 0 kolom<br>
jumlah kolom object sebanyak 1 kolom<br>

In [161]:
df.describe()

myct          mmin          mmax        cach       chmin  \
count   209.000000    209.000000    209.000000  209.000000  209.000000   
mean    203.822967   2867.980861  11796.153110   25.205742    4.698565   
std     260.262926   3878.742758  11726.564377   40.628722    6.816274   
min      17.000000     64.000000     64.000000    0.000000    0.000000   
25%      50.000000    768.000000   4000.000000    0.000000    1.000000   
50%     110.000000   2000.000000   8000.000000    8.000000    2.000000   
75%     225.000000   4000.000000  16000.000000   32.000000    6.000000   
max    1500.000000  32000.000000  64000.000000  256.000000   52.000000   

            chmax          prp  
count  209.000000   209.000000  
mean    18.267943   105.622010  
std     25.997318   160.830733  
min      0.000000     6.000000  
25%      5.000000    27.000000  
50%      8.000000    50.000000  
75%     24.000000   113.000000  
max    176.000000  1150.000000

In [171]:
df.head()

vendor  myct  mmin   mmax  cach  chmin  chmax  prp
0       0   125   256   6000   256     16    128  198
1       1    29  8000  32000    32      8     32  269
2       1    29  8000  32000    32      8     32  220
3       1    29  8000  32000    32      8     32  172
4       1    29  8000  16000    32      8     16  132

In [162]:
df2 = df.copy()

In [163]:
#mengubah fitur vendor dari data text menjadi data numerik
from sklearn.preprocessing import LabelEncoder
LabelEncode = LabelEncoder()
df['vendor'] = LabelEncode.fit_transform(df['vendor'])
df.head()

vendor  myct  mmin   mmax  cach  chmin  chmax  prp
0       0   125   256   6000   256     16    128  198
1       1    29  8000  32000    32      8     32  269
2       1    29  8000  32000    32      8     32  220
3       1    29  8000  32000    32      8     32  172
4       1    29  8000  16000    32      8     16  132

In [ ]:
from sklearn.preprocessing import LabelEncoder

categoric = ["cylinders", "model_year", "origin", "car_name"]

encode = LabelEncoder()
df2[categoric] = encode.fit_transform(df[categoric])
df2 = pd.DataFrame(df2, columns=categoric)
df2.head()

In [164]:
from sklearn.preprocessing import MinMaxScaler

numeric = ["myct", "mmin", "mmax", "cach", "chmin", "prp"]

scaller = MinMaxScaler()
df2 = scaller.fit_transform(df2[numeric])
df2 = pd.DataFrame(df2, columns=numeric)
df2.head()

myct      mmin      mmax   cach     chmin       prp
0  0.072825  0.006012  0.092843  1.000  0.307692  0.167832
1  0.008092  0.248497  0.499499  0.125  0.153846  0.229895
2  0.008092  0.248497  0.499499  0.125  0.153846  0.187063
3  0.008092  0.248497  0.499499  0.125  0.153846  0.145105
4  0.008092  0.248497  0.249249  0.125  0.153846  0.110140

In [165]:
df2['vendor'] = df['vendor']
df2.head()

myct      mmin      mmax   cach     chmin       prp  vendor
0  0.072825  0.006012  0.092843  1.000  0.307692  0.167832       0
1  0.008092  0.248497  0.499499  0.125  0.153846  0.229895       1
2  0.008092  0.248497  0.499499  0.125  0.153846  0.187063       1
3  0.008092  0.248497  0.499499  0.125  0.153846  0.145105       1
4  0.008092  0.248497  0.249249  0.125  0.153846  0.110140       1

In [166]:
X = df.drop(columns=["prp"])
y = df["prp"]

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

data testing 20%<br>
data training 80%<br>
tidak memakai stratified karena bukan merupakan klasifikasi

In [167]:
from sklearn.linear_model import LinearRegression
model = LinearRegression()
model.fit(X_train, y_train)

print(model.score(X_train, y_train), model.score(X_test, y_test))

0.8619691625301995 0.8806001439672928


In [210]:
hasil = model.predict(X_test)
from sklearn.metrics import mean_squared_error, r2_score

print('MSE = ', mean_squared_error(y_test, hasil), '\nR2 = ', r2_score(y_test, hasil))

MSE =  0.6206896551724138 
R2 =  -0.03882706522622037


# Risk Level

In [174]:
df = pd.read_csv("Maternal Health Risk Data Set.csv")
df.head()

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel
0   25         130           80  15.0      98.0         86  high risk
1   35         140           90  13.0      98.0         70  high risk
2   29          90           70   8.0     100.0         80  high risk
3   30         140           85   7.0      98.0         70  high risk
4   35         120           60   6.1      98.0         76   low risk

In [175]:
df.describe()

Age   SystolicBP  DiastolicBP           BS     BodyTemp  \
count  1014.000000  1014.000000  1014.000000  1014.000000  1014.000000   
mean     29.871795   113.198225    76.460552     8.725986    98.665089   
std      13.474386    18.403913    13.885796     3.293532     1.371384   
min      10.000000    70.000000    49.000000     6.000000    98.000000   
25%      19.000000   100.000000    65.000000     6.900000    98.000000   
50%      26.000000   120.000000    80.000000     7.500000    98.000000   
75%      39.000000   120.000000    90.000000     8.000000    98.000000   
max      70.000000   160.000000   100.000000    19.000000   103.000000   

         HeartRate  
count  1014.000000  
mean     74.301775  
std       8.088702  
min       7.000000  
25%      70.000000  
50%      76.000000  
75%      80.000000  
max      90.000000

In [176]:
df.shape

(1014, 7)

In [178]:
df.isna().sum()

Age            0
SystolicBP     0
DiastolicBP    0
BS             0
BodyTemp       0
HeartRate      0
RiskLevel      0
dtype: int64

In [179]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1014 entries, 0 to 1013
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Age          1014 non-null   int64  
 1   SystolicBP   1014 non-null   int64  
 2   DiastolicBP  1014 non-null   int64  
 3   BS           1014 non-null   float64
 4   BodyTemp     1014 non-null   float64
 5   HeartRate    1014 non-null   int64  
 6   RiskLevel    1014 non-null   object 
dtypes: float64(2), int64(4), object(1)
memory usage: 55.6+ KB


In [180]:
df.dtypes

Age              int64
SystolicBP       int64
DiastolicBP      int64
BS             float64
BodyTemp       float64
HeartRate        int64
RiskLevel       object
dtype: object

In [181]:
df.head()

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel
0   25         130           80  15.0      98.0         86  high risk
1   35         140           90  13.0      98.0         70  high risk
2   29          90           70   8.0     100.0         80  high risk
3   30         140           85   7.0      98.0         70  high risk
4   35         120           60   6.1      98.0         76   low risk

In [182]:
from sklearn.preprocessing import MinMaxScaler

numeric = ["Age", "SystolicBP", "DiastolicBP", "BS", "BodyTemp", "HeartRate"]
scaler = MinMaxScaler()

df_numeric = scaller.fit_transform(df[numeric])
df_numeric = pd.DataFrame(df_numeric, columns=numeric)
df_numeric.head()

Age  SystolicBP  DiastolicBP        BS  BodyTemp  HeartRate
0  0.250000    0.666667     0.607843  0.692308       0.0   0.951807
1  0.416667    0.777778     0.803922  0.538462       0.0   0.759036
2  0.316667    0.222222     0.411765  0.153846       0.4   0.879518
3  0.333333    0.777778     0.705882  0.076923       0.0   0.759036
4  0.416667    0.555556     0.215686  0.007692       0.0   0.831325

In [183]:
from sklearn.preprocessing import LabelEncoder

encoding = LabelEncoder()
df['RiskLevel'] = encoding.fit_transform(df['RiskLevel'])
df.head()

Age  SystolicBP  DiastolicBP    BS  BodyTemp  HeartRate  RiskLevel
0   25         130           80  15.0      98.0         86          0
1   35         140           90  13.0      98.0         70          0
2   29          90           70   8.0     100.0         80          0
3   30         140           85   7.0      98.0         70          0
4   35         120           60   6.1      98.0         76          1

In [185]:
df_numeric['RiskLevel'] = df['RiskLevel']
df_numeric.head()

Age  SystolicBP  DiastolicBP        BS  BodyTemp  HeartRate  RiskLevel
0  0.250000    0.666667     0.607843  0.692308       0.0   0.951807          0
1  0.416667    0.777778     0.803922  0.538462       0.0   0.759036          0
2  0.316667    0.222222     0.411765  0.153846       0.4   0.879518          0
3  0.333333    0.777778     0.705882  0.076923       0.0   0.759036          0
4  0.416667    0.555556     0.215686  0.007692       0.0   0.831325          1

In [188]:
X = df.drop(columns=["RiskLevel"])
y = df["RiskLevel"]

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0, stratify=y)

In [189]:
df['RiskLevel'].value_counts()

1    406
2    336
0    272
Name: RiskLevel, dtype: int64

In [204]:
from sklearn.svm import SVC
model = SVC(random_state=0, decision_function_shape='ovo', kernel='linear')
model.fit(X_train, y_train)

print(model.score(X_train, y_train), model.score(X_test, y_test))

0.6633785450061652 0.6748768472906403


In [205]:
hasil = model.predict(X_test)
from sklearn.metrics import classification_report

print(classification_report(y_test, hasil))

              precision    recall  f1-score   support

           0       0.76      0.76      0.76        55
           1       0.67      0.86      0.75        81
           2       0.58      0.37      0.45        67

    accuracy                           0.67       203
   macro avg       0.67      0.67      0.66       203
weighted avg       0.66      0.67      0.66       203

